src/v/raft/notification.h

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "utils/named_type.h"
+
+namespace raft {
+using notification_id_type = named_type<int32_t, struct raft_notification_id>;
```

Comment:
I'm not a fan of reusing these types across multiple notifying classes, even in the same submodule. Why not define them inside the notifying class, e.g. this type would be `raft::group_manager::notification_id`? This way it is never confused with a notification id from any other class?

Reply:
agreed, but there are also some caveats, all the components would have to include on the `group_manager` type header to access the type. Maybe we can meet in the middle and create a separate per notification type in the `notifications.h` header ?

Reply:
I'd rather do it in `utils/notification_list.h`. Something like this:

```
template<class Tags...>
struct notification_id_tag;

template<class Tags...>
class notification_id_type: public named_type<int32_t, notification_id_tag<Tags...>> {
    static notification_id_type invalid {-1};
    // todo default ctor to init as invalid
    // todo ctor template to call base ctor with a forwarded argument
}
```

and then use it like `notification_id_type<raft::group_manager>`.

This may be too much for this PR though. If you think it is maybe just tag this type more specifically, not with `struct raft_notification_id` but rather with e.g. `struct group_manager_notification_id`.